src/components/Metrics/TrafficDetails.tsx

```diff
@@ -0,0 +1,221 @@
+    const isAppSet =
+      prevProps.itemType === MetricsObjectTypes.APP &&
+      this.props.itemType === prevProps.itemType &&
+      (prevProps.namespace !== this.props.namespace || prevProps.appName !== this.props.appName);
+
+    if (isWorkloadSet || isAppSet || prevProps.trafficData !== this.props.trafficData) {
+      this.processTrafficData(this.props.trafficData);
+    }
+  }
+
+  render() {
+    const rateIntervalName = getName(this.props.rateInterval).toLowerCase();
+
+    return (
+      <Row className="card-pf-body">
+        <Col xs={12}>
+          <div>
+            <strong>Inbound ({rateIntervalName})</strong>
+          </div>
+          <DetailedTrafficList direction="inbound" traffic={this.state.inboundTraffic} />
+          <div style={{ marginTop: '2em' }}>
+            <strong>Outbound ({rateIntervalName})</strong>
+          </div>
+          <DetailedTrafficList direction="outbound" traffic={this.state.outboundTraffic} />
+        </Col>
+      </Row>
+    );
+  }
+
+  private buildTrafficNode = (prefix: 'in' | 'out', node: any): TrafficNode => {
+    if (this.props.itemType === MetricsObjectTypes.WORKLOAD) {
+      return {
+        id: `${prefix}-${node.id}`,
+        type: node.nodeType,
+        namespace: node.namespace,
+        name: node.workload
+      };
+    } else {
+      return {
+        id: `${prefix}-${node.id}`,
+        type: node.nodeType,
+        namespace: node.namespace,
+        name: node.app,
+        version: node.version
+      };
+    }
+  };
+
+  private processTrafficSiblings = (edges: any, serviceTraffic: ServiceTraffic, nodes: any, myNode: any) => {
+    const inboundTraffic: TrafficItem[] = [];
+    const outboundTraffic: TrafficItem[] = [];
+
+    edges.forEach(edge => {
+      const sourceNode = nodes['id-' + edge.data.source];
+      const targetNode = nodes['id-' + edge.data.target];
+
+      if (myNode.id === edge.data.source || myNode.id === edge.data.target) {
+        return;
+      }
+
+      if (targetNode.nodeType === NodeType.SERVICE) {
+        const svcId = `in-${targetNode.namespace}-${targetNode.service}`;
+        if (serviceTraffic[svcId]) {
+          inboundTraffic.push({
+            traffic: edge.data.traffic,
+            proxy: serviceTraffic[svcId],
+            node: this.buildTrafficNode('in', sourceNode)
+          });
+        }
+      } else if (sourceNode.nodeType === NodeType.SERVICE) {
+        const svcId = `out-${sourceNode.namespace}-${sourceNode.service}`;
+        if (serviceTraffic[svcId]) {
+          outboundTraffic.push({
+            traffic: edge.data.traffic,
+            proxy: serviceTraffic[svcId],
+            node: this.buildTrafficNode('out', targetNode)
+          });
+        }
+      }
+    });
+
+    return { inboundTraffic, outboundTraffic };
+  };
+
+  private processServicesTraffic = (edges: any, nodes: any, myNode: any) => {
```

Comment:
that makes a lot of `any`, it makes it difficult to understand how the data is structured :-(